### PR TITLE
udpating gitlab initialize route

### PR DESCRIPTION
**Pull request by ric03uec**
- https://github.com/Shippable/admiral/issues/97
- https://github.com/Shippable/admiral/issues/99

### CODE REVIEW

File: api/state/post.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        );
+      }
+      fs.chmodSync(bag.tmpScript, '755');
+      return next();
+    }
+  );
+}
+
+
+function _initializeMsg(bag, next) {
```
Comment (trriplejay) [inline api/state/post.js]: should this be `_intializeState`?

File: api/state/post.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+function _post(bag, next) {
+  var who = bag.who + '|' + _post.name;
+  logger.verbose(who, 'Inside');
+
+  var update = bag.config;
+  bag.config.isInstalled = true;
+  bag.config.isInitialized = true;
+  bag.config.rootPassword = bag.reqBody.password;
+
+  var query = util.format('UPDATE "systemConfigs" set state=\'%s\';',
```
Comment (trriplejay) [inline api/state/post.js]: should this also use the `bag.component` var that was set up at the top?

File: common/scripts/docker/installState.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+export COMPONENT="state"
+export STATE_DATA_DIR="$RUNTIME_DIR/$COMPONENT/data"
+export STATE_LOGS_DIR="$RUNTIME_DIR/$COMPONENT/logs"
+export STATE_CONFIG_DIR="$CONFIG_DIR/$COMPONENT"
+export SCRIPTS_DIR="$SCRIPTS_DIR"
+export STATE_IMAGE="gitlab/gitlab-ce:8.9.6-ce.0"
+export STATE_PORT=80
+
+__validate_state_envs() {
+  __process_msg "Initializing vault environment variables"
```
Comment (trriplejay) [inline common/scripts/docker/installState.sh]: says 'vault' still.

File: common/scripts/docker/installState.sh [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  __process_msg "SCRIPTS_DIR: $SCRIPTS_DIR"
+  __process_msg "STATE_DATA_DIR: $STATE_DATA_DIR"
+  __process_msg "MSGSTATE_CONFIG_DIR: $STATE_CONFIG_DIR"
```
Comment (a-murphy) [inline common/scripts/docker/installState.sh]: What is this `MSGSTATE`?

File: common/scripts/docker/installState.sh [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  fi
+
+  if [ ! -d "$STATE_CONFIG_DIR" ]; then
+    __process_msg "Creating config directory $STATE_CONFIG_DIR"
+    sudo mkdir -p $STATE_CONFIG_DIR
+  else
+    __process_msg "Config directory already present: $STATE_CONFIG_DIR"
+  fi
+
+  sudo mkdir -p $MSG_CONFIG_DIR/scripts
```
Comment (a-murphy) [inline common/scripts/docker/installState.sh]: And is this defined?

**Pull request merged**